common/src/unifyfs_client_rpcs.h

```diff
@@ -288,26 +288,28 @@ DECLARE_MARGO_RPC_HANDLER(unifyfs_heartbeat_rpc)
  * returns node local extents for a file
  * given a global file id */
 MERCURY_GEN_STRUCT_PROC(unifyfs_client_index_t,
-                 ((uint32_t)(file_pos))
-                 ((uint32_t)(log_pos))
-                 ((uint32_t)(length))
+                 ((uint64_t)(file_pos))
+                 ((uint64_t)(log_pos))
+                 ((uint64_t)(length))
                  ((int32_t)(gfid))
                  ((int32_t)(log_app_id))
                  ((int32_t)(log_client_id)))
 /* List of extents used in the output of RPC call*/
 typedef struct extents_list {
     unifyfs_client_index_t value;
     struct extents_list* next;
-} *extents_list_t;
+} extents_list_t;
+
+typedef extents_list_t* extents_list_t_ptr;
```

Comment:
However, the macros don't work with a pointer so typedef is still needed over a pointer. I only use this here and nowhere else.